src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala

```diff
@@ -104,10 +107,15 @@ class PeerManagerActor(
       stash()
 
     case (HandlePeerConnection(connection, remoteAddress), Success(_)) =>
-      context.unbecome()
-      val peer = createPeer(remoteAddress)
-      peer.ref ! PeerActor.HandleConnection(connection, remoteAddress)
-      unstashAll()
+      if (!isConnectionHandled(remoteAddress)) {
+        context.unbecome()
+        val peer = createPeer(remoteAddress)
+        peer.ref ! PeerActor.HandleConnection(connection, remoteAddress)
+        unstashAll()
+      } else {
+        log.info("Another connection with {} is already opened. Disconnecting.", remoteAddress)
+        connection ! PoisonPill
+      }
 
     case (HandlePeerConnection(connection, remoteAddress), Failure(_)) =>
```

Comment:
As in this case we are also establishing a connection with a new peer, should we also check whether  `isConnectionHandled` is true or not?